test/unit/org/openstreetmap/josm/gui/dialogs/properties/PropertiesDialogFilterActionTest.java

```diff
@@ -0,0 +1,197 @@
+import org.openstreetmap.josm.gui.dialogs.properties.PropertiesDialog.ReadOnlyTableModel;
+import org.openstreetmap.josm.gui.layer.OsmDataLayer;
+import org.openstreetmap.josm.testutils.JOSMTestRules;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+/**
+ * Tests of {@link PropertiesDialog.FilterAction} class
+ */
+public class PropertiesDialogFilterActionTest {
+
+    /**
+     * Setup tests
+     */
+    @RegisterExtension
+    @SuppressFBWarnings(value = "URF_UNREAD_PUBLIC_OR_PROTECTED_FIELD")
+    public JOSMTestRules test = new JOSMTestRules().main().projection().preferences();
+
+    @Test
+    void createFilterForKeysTest() {
+        List<OsmPrimitive> prims = new ArrayList<>();
+        Node n = new Node();
+        n.put("foo", "bar");
+        Way w = new Way();
+        w.put("foo", "baz");
+        w.put("blah", "blub");
+        prims.add(n);
+        prims.add(w);
+
+        Filter filter = PropertiesDialog.createFilterForKeys(Arrays.asList("foo", "blah"), prims);
+        SearchSetting expectedSearchSetting = new SearchSetting();
+        expectedSearchSetting.text = "((\"foo\"=\"bar\") OR (\"foo\"=\"baz\")) AND (\"blah\"=\"blub\")";
+        expectedSearchSetting.caseSensitive = true;
+        Filter expected = new Filter(expectedSearchSetting);
+        assertEquals(0, filter.compareTo(expected));
+    }
+
+    @Test
+    void filterActionHideTest() {
+        DataSet ds = new DataSet();
+        OsmDataLayer layer = new OsmDataLayer(ds, "", null);
+        MainApplication.getLayerManager().addLayer(layer);
+        MapFrame mf = MainApplication.getMap();
+        PropertiesDialog propertiesDialog = mf.propertiesDialog;
+        FilterDialog filterDialog = mf.filterDialog;
+        FilterTableModel filterModel = filterDialog.getFilterModel();
+
+        Node n = new Node(new EastNorth(0, 0));
+        n.put("natural", "tree");
+        n.put("leaf_type", "broadleaved");
+        ds.addPrimitive(n);
+
+        Node n2 = new Node(new EastNorth(0, 0));
+        n2.put("natural", "tree");
+        n2.put("leaf_type", "needleleaved");
+        ds.addPrimitive(n2);
+
+        Node n3 = new Node(new EastNorth(0, 0));
+        n3.put("natural", "tree");
+        ds.addPrimitive(n3);
+
+        try {
+            Field filterActionHideField = PropertiesDialog.class.getDeclaredField("filterActionHide");
+            filterActionHideField.setAccessible(true);
+            FilterAction filterActionHide = (PropertiesDialog.FilterAction) filterActionHideField.get(propertiesDialog);
+
+            Field tagTableField = PropertiesDialog.class.getDeclaredField("tagTable");
+            tagTableField.setAccessible(true);
+            JTable tagTable = (JTable) tagTableField.get(propertiesDialog);
+
+            Field tagDataField = PropertiesDialog.class.getDeclaredField("tagData");
+            tagDataField.setAccessible(true);
+            ReadOnlyTableModel tagData = (ReadOnlyTableModel) tagDataField.get(propertiesDialog);
```

Comment:
Maybe this should be extracted? The only difference between this block and the block in `filterActionShowOnlyTest` is the `filterActionHide` field.